### PR TITLE
brew.sh: use HOMEBREW_GIT to detect HOMEBREW_VERSION

### PR DESCRIPTION
**Pull request by maxim-belkin**
Move things a tiny bit so that we can use `HOMEBREW_GIT` (instead of `git`) to detect `HOMEBREW_VERSION`.

-----

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Review by claui — APPROVED** [review]
Makes sense. LGTM

File: Library/Homebrew/brew.sh [hunk 1]
```diff
@@ -144,6 +136,14 @@ else
   HOMEBREW_GIT="git"
 fi
 
+HOMEBREW_VERSION="$("$HOMEBREW_GIT" -C "$HOMEBREW_REPOSITORY" describe --tags --dirty --abbrev=7 2>/dev/null)"
+HOMEBREW_USER_AGENT_VERSION="$HOMEBREW_VERSION"
+if [[ -z "$HOMEBREW_VERSION" ]]
+then
+  HOMEBREW_VERSION=">=2.2.0 (shallow or no git repository)"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/brew.sh:143]:
```suggestion
  HOMEBREW_VERSION=">=2.5.0 (shallow or no git repository)"
```

**Review by MikeMcQuaid — APPROVED** [review]
Fine with me! One suggestion given we're bumping anyway.

**Comment by MikeMcQuaid** [comment]
Thanks @maxim-belkin!

**Pull request merged**